protobuf-serde/src/lib.rs

```diff
@@ -0,0 +1,5 @@
+pub fn serialize_oneof () {
```

Comment:
I assume this is just a stub yet which you are going to implement. Could you please add TODO next time to avoid confusion?

BTW, `oneof` can be serialized with `derive`. Custom serializers are needed for `RepeatedField` etc.

Reply:
If we serialize a `OneOf` it looks like it'll include the `value` and `set` fields whereas we just want the `value` so I'm thinking we'll want a custom serializer here

So that we end up with

```
{ "oneOfField": {"datatype": {}} } 
```

Instead of

```
{"oneOfField": {"value": {"datatype": {}", "set": true}}}
```

Reply:
> `{"oneOfField": {"value": {"datatype": {}", "set": true}}}`

Is it serde adding `set` field? Why?

What is `datatype`? Could you please post a protobuf schema for this JSON generated, because it's hard to reply without understanding context?

Anyway

* it's not possible to implement oneof serializer in `protobuf-serde`, protobuf type system won't allow it
* this is exactly the reason why I think JSON mapping should be implemented natively

Reply:
Now I get it. It's singular field (`optional` or `repeated`), not `oneof`.

Reply:
Ah apologies for the confusion - yeah it's this:

https://github.com/stepancheg/rust-protobuf/blob/d153996e9354abe3c99933a3224288064056ae30/protobuf-test/src/common/v2/test_serde_derive_pb.proto#L10

Guess I'll need to add a test for an inlined `oneof` also after I get this working.